otherlibs/unix/link.c

```diff
@@ -18,18 +18,32 @@
 #include <caml/signals.h>
 #include "unixsupport.h"
 
-CAMLprim value unix_link(value path1, value path2)
+#include <fcntl.h>
+#include <unistd.h>
+
+CAMLprim value unix_link(value follow, value path1, value path2)
 {
-  CAMLparam2(path1, path2);
+  CAMLparam3(follow, path1, path2);
   char * p1;
   char * p2;
   int ret;
   caml_unix_check_path(path1, "link");
   caml_unix_check_path(path2, "link");
   p1 = caml_strdup(String_val(path1));
   p2 = caml_strdup(String_val(path2));
+  ret =
+    Is_block(follow) && Bool_val(Field(follow, 0)) /* Some true */
+    ? AT_SYMLINK_FOLLOW
+    : 0;
   caml_enter_blocking_section();
-  ret = link(p1, p2);
+  if (follow == Val_int(0) /* None */)
+    ret = link(p1, p2);
+  else /* Some bool */
+# if _XOPEN_VERSION >= 700 || _POSIX_VERSION >= 200809L
```

Comment:
ok. I included the change in my tree. But I have to admit I don't understand [feature_test_macros(7)](https://linux.die.net/man/7/feature_test_macros). Aren't those macros used to _request_, not _check_ for features?